### PR TITLE
Feat/#12 handle state

### PR DESCRIPTION
**Pull request by teatimes**

### CODE REVIEW

File: reducers/sessions.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+const removeExercise = (state, action) => {
+  let activeSession = state.activeSession
+  activeSession.exercises = activeSession.exercises.filter(exercise => exercise.name != action.payload.name )
+  return { ...state, activeSession: activeSession }
+}
+
+// If no state is given, state is set to INITIAL_STATE
+export default function (state = INITIAL_STATE, action) {
+  switch (action.type) {
+    case 'ADD_SESSION':
```
Comment (Mariusun) [inline reducers/sessions.js:70]: Should import action types from actions/types.js

File: src/components/AddContact.js [hunk 2]
```diff
@@ -14,6 +17,19 @@ export default class AddContact extends Component {
...
+    this.availableContacts = this.availableContacts.bind(this)
+  }
+
+  _add(contact) {
+    this.props.addContact(contact)
+    this.props.navigation.goBack()
+  }
+
+  // Filter out the contacts allready added, so that they can't be added twice
+  availableContacts() {
```
Comment (Mariusun) [inline src/components/AddContact.js:33]: This function is unused. Component currently has no prop contacts.

File: src/components/CreateSession.js [hunk 3]
```diff
@@ -18,9 +21,11 @@ class CreateSession extends Component {
     super(props, context)
     // If navigation prop has params, set the propDate in params as propDate
     let propDate = this.props.navigation.state.params != null ? this.props.navigation.state.params.date : null
+
     this.state = {
       // If propDate isn't null, set it as date. If it is, set today as date
       date: propDate != null ? propDate : new Date(),
```
Comment (Mariusun) [inline src/components/CreateSession.js]: state.date and selectedDate from the store seem to store the same? Should only be in the store when of interest to other components, and be included as prop for the component in mapStateToProps.

File: src/components/__tests__/AddContact.spec.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+describe('AddContact', () => {
+  let wrapper
+  let dispatchAddContactMock = jest.fn()
+  let navigation
+  let goBackMock
+
+  beforeEach(() => {
+    goBackMock = jest.fn()
+    navigation = {goBack: goBackMock}
+    wrapper = shallow(<AddContact navigation={navigation} addExercise={dispatchAddContactMock} addContact={dispatchAddContactMock} />)
```
Comment (Mariusun) [inline src/components/__tests__/AddContact.spec.js]: AddContact should not be given addExercise as prop

**Review by Mariusun — CHANGES_REQUESTED** [review]
Errors:
Choose not today's date, then create session, then add contact
Choose not today's date, then create session, then add exercise

Bugs:
Start fresh, click today's date -> session changes, but shouldn't
Newly created sessions do not show up

File: reducers/sessions.js (hunk 1 above)
Comment (teatimes) [reply reducers/sessions.js:70]: Import and use how, you mean?

File: src/components/AddContact.js (hunk 2 above)
Comment (teatimes) [reply src/components/AddContact.js:33]: The method is not implemented yet, just copied it from earlier so we don't have to rewrite it. Have commented it out for now

File: reducers/sessions.js (hunk 1 above)
Comment (Mariusun) [reply reducers/sessions.js:70]: Like in the reducers in e.g. this project: https://github.com/gothinkster/react-redux-realworld-example-app/blob/master/src/reducers/common.js

Comment (teatimes) [reply reducers/sessions.js:70]: Is there any important reason to do it that way instead? It works as it is now, and I think I tried to do what you suggests, but then it just failed

Comment (Mariusun) [reply reducers/sessions.js:70]: It's good practice, but if you encountered a problem just leave it as is.

**Review by Mariusun — APPROVED** [review]

**Pull request merged**